streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -59,6 +60,8 @@
     private final File baseDir;
     private final FileLock directoryLock;
     private final Map<String, StateStore> stores;
+    private final Map<String, ProcessorContext> storeContext;
```

Comment:
Since a `AbstractTask` contains exactly one `ProcessorStateManager` and one `ProcessorContext`, we can just pass the context from `AbstractTask' to the constructor of ProcessorStateManager and remove this map.